webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala

```diff
@@ -362,6 +364,12 @@ class Boot extends Loggable {
     // the session itself.
     LiftRules.noCometSessionCmd.default.set(() => JsCmd.unitToJsCmd(()))
 
+    // Log CSP violations
+    LiftRules.contentSecurityPolicyViolationReport = (r: ContentSecurityPolicyViolation) => {
+      ApplicationLogger.warn(s"Content security policy violation: blocked ${r.blockedUri} in ${r.documentUri} because of ${r.violatedDirective} directive")
```

Comment:
I haven't investigated how to log is the right context, Application looks a bit too generic.

Reply:
Seems correct to me